server/src/main/java/org/elasticsearch/script/ScriptContext.java

```diff
@@ -96,13 +99,21 @@ public ScriptContext(String name, Class<FactoryType> factoryClazz, int cacheSize
         this.cacheSizeDefault = cacheSizeDefault;
         this.cacheExpireDefault = cacheExpireDefault;
         this.maxCompilationRateDefault = maxCompilationRateDefault;
+        this.allowStoredScript = allowStoredScript;
+    }
+
+    /** Construct a context with the related instance and compiled classes with caller provided cache defaults and allow
+     * scripts of this context to be stored scripts */
+    public ScriptContext(String name, Class<FactoryType> factoryClazz, int cacheSizeDefault, TimeValue cacheExpireDefault,
```

Comment:
I do not think we should add a new ctor. Instead, we should update the existing uses with the new parameter. I would separate this change out into its own PR.